controllers/front/action.php

```diff
@@ -17,25 +18,366 @@
     {
-        dump($params);
+        if (
+            isset($params['idWishlist'])
+            && isset($params['id_product'])
+            && isset($params['id_product_attribute'])
+            && isset($params['id_customer'])
+        ) {
+            $isDeleted = wishlist::removeProduct(
+                $params['idWishlist'],
+                $params['id_customer'],
+                $params['id_product'],
+                $params['id_product_attribute']
+            );
+
+            if (true === $isDeleted) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'success' => true,
+                        'message' => $this->module->l('Product succesfully removed', 'mywishlist')
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('Unable to remove product from wishlist', 'mywishlist')
+                ])
+            );
+        }
+
+        $this->ajaxDieMissingParams();
+    }
+
+    public function updateProductFromWishlistAction($params)
+    {
+        if (isset(
+            $params['idWishlist'],
+            $params['id_product'],
+            $params['id_product_attribute'],
+            $params['priority'],
+            $params['quantity']
+        )) {
+            $isDeleted = wishlist::updateProduct(
+                $params['idWishlist'],
+                $params['id_product'],
+                $params['id_product_attribute'],
+                $params['priority'],
+                $params['quantity']
+            );
+
+            if ($isDeleted) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'success' => true,
+                        'message' => $this->module->l('Product succesfully updated', 'mywishlist')
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('Unable to update product from wishlist', 'mywishlist')
+                ])
+            );
+        }
+
+        $this->ajaxDieMissingParams();
+    }
+
+    public function getAllWishlistAction()
+    {
+        $infos = Wishlist::getAllWishlistsByIdCustomer($this->context->customer->id);
+
+        if (!empty($infos)) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'wishlists' => $infos,
+                ])
+            );
+        }
+        $this->ajaxDieMissingParams();
+    }
+
+    private function generateWishlistToken()
+    {
+        return strtoupper(substr(sha1(uniqid(rand(), true) . _COOKIE_KEY_ . $this->context->customer->id), 0, 16));
+    }
+
+    private function ajaxDieMissingParams()
+    {
+        $this->ajaxDie(
+            Tools::jsonEncode([
```

Comment:
```suggestion
            json_encode([
```
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/Tools.php#L3007